server/NetPacksLobbyServer.cpp

```diff
@@ -118,26 +118,17 @@ void ClientPermissionsCheckerNetPackVisitor::visitLobbyChatMessage(LobbyChatMess
 
 void ApplyOnServerNetPackVisitor::visitLobbyQuickLoadGame(LobbyQuickLoadGame & pack)
 {
+	srv.prepareToRestart();
 	// modify StartInfo to load the quicksave
 	srv.si->mode = EStartMode::LOAD_GAME;
 	srv.si->mapname = pack.saveFilePath;
-
-	// prepare game state (loads the save file)
-	if (!srv.prepareToStartGame()) {
-		//failure is destructive and an exception is the only way
-		throw std::runtime_error("Failed to prepare to start game during quick load.");
-	}
-
-	// create LobbyStartGame packet with loaded state and announce to all clients
-	LobbyStartGame startPack;
-	startPack.initializedStartInfo = std::make_shared<StartInfo>(*srv.gh->gameState().getInitialStartInfo());
-	startPack.initializedGameState = srv.gh->gs;
-	srv.announcePack(startPack);
```

Comment:
No longer needed? How clients receive correct initial gamestate now?

Reply:
The two lines above suffice:
```
	// modify StartInfo to load the quicksave
	srv.si->mode = EStartMode::LOAD_GAME;
	srv.si->mapname = pack.saveFilePath;
```
This follows the same logic as restart game, where StartInfo is set and then the StartGame logic takes care of the rest, which is initiated later by the added lines in `client/NetPacksLobbyClient.cpp`
```
	handler.restartGameplay();
	handler.sendStartGame();
```
I just now finished a three hour hotseat session with this PR code and it works well. Had not a single crash, yay, but some other vcmi bugs 🥲.